timeplot/timelines.py

```diff
@@ -0,0 +1,123 @@
+from matplotlib.lines import Line2D
+import matplotlib
+
+matplotlib.use("agg")
+
+
+DEFAULT_COLORS = [
+    "#1b9e77",
+    "#d95f02",
+    "#7570b3",
+    "#e7298a",
+    "#66a61e",
+    "#e6ab02",
+    "#a6760d",
+    "#666666",
+]
+
+
+def _make_default_color_lookup(event_types):
+    """
+    If no colors are supplied, create a default color mapping for event types.
+
+    Parameters
+    ----------
+    event_types : array-like
+        The event types observed in the dataset.
+
+    Returns
+    -------
+    dict
+    """
+    n_events = len(event_types)
+    if n_events > len(DEFAULT_COLORS):
+        raise LookupError(
+            f"There are more event types ({n_events}) than default colors "
+            + f"({len(DEFAULT_COLORS)}). Please provide a custom color lookup."
+        )
+
+    return {event_types[i]: DEFAULT_COLORS[i] for i in range(n_events)}
+
+
+def plot_events_timelime(
+    df,
+    ax=None,
+    custom_color_lookup=None,
+    show_y_axis=False,
+    **kwargs,
+):
+    """
+    Make plots event series.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        The data to plot.
```

Comment:
This makes it sounds like it can plot any data. From the code, it doesn't look like that's the case. This would be more helpful if it declared the columns it's going to grab and any datatype expectations it has (I'm guessing it won't work with strings).